drivers/gles3/rasterizer_scene_gles3.h

```diff
@@ -817,6 +817,20 @@ class RasterizerSceneGLES3 : public RasterizerScene {
 		}
 	};
 
+	enum class OcclusionQueryPrevAction {
+		NoAction,
+		Rendered,
+		Culled
+	};
+	struct OcclusionQueryData {
+		OcclusionQueryPrevAction prev_action = OcclusionQueryPrevAction::NoAction;
+		uint64_t prev_frame = 0;
+		GLuint query;
+	};
+	//TODO: remove stuff from this map when an instance is removed from the rendering server.
```

Comment:
I didn't notice where you do this cleanup, but indeed, you will need to ensure that you aren't leaking queries